app/src/sharedTest/java/org/oppia/app/profile/AdminPinActivityTest.kt

```diff
@@ -0,0 +1,179 @@
+package org.oppia.app.profile
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import org.hamcrest.Matchers.allOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.domain.profile.ProfileTestHelper
+import org.oppia.util.logging.EnableConsoleLog
+import org.oppia.util.logging.EnableFileLog
+import org.oppia.util.logging.GlobalLogLevel
+import org.oppia.util.logging.LogLevel
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+class AdminPinActivityTest {
+
+  @Inject lateinit var context: Context
+  @Inject lateinit var profileTestHelper: ProfileTestHelper
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    profileTestHelper.initializeProfiles()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
```

Comment:
To maintain code order, we should shift this function after all test cases, which we have followed in all other test files.